File: android/app/src/main/java/com/zahraag/pawsitivehabits/data/models/Weight.kt
package com.zahraag.pawsitivehabits.data.models

import androidx.room.Entity
import androidx.room.PrimaryKey
import com.google.gson.annotations.SerializedName
import kotlinx.serialization.Serializable
import java.util.UUID

@Serializable
@Entity(tableName = "weight_table")
data class Weight(
    @PrimaryKey @SerializedName("_id")
    val id: String = org.bson.types.ObjectId().toHexString(),
    var userId: String,
    var petId: String,
    var weightValue: Double,
    var unit: String,
    var date: Long,

    @Transient
    var isSynced: Boolean = false
)

File: api/src/models/weight.js
const mongoose = require('mongoose');

const WeightSchema = new mongoose.Schema({
    _id: { type: String, required: true},
    userId: {
            type: mongoose.Schema.Types.ObjectId,
            ref: 'User',
            required: true,
            index: true
          },
    petId: { type: String,
            ref: 'Pet',
            required: true,
            index: true },
    weightValue: { type: Number, required:true},
    unit: { type: String ,default: 'kg'},
    date: { type: Number, required: true }
}, { timestamps: true });

module.exports = mongoose.model('Weight', WeightSchema);